cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -434,14 +417,58 @@ func del(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 	return nil
 }
 
-func cleanUpPodENI(podVlanId int, log logger.Logger, containerId string, driverClient driver.NetworkAPIs) error {
-	err := driverClient.TeardownPodENINetwork(podVlanId, log)
-	if err != nil {
-		log.Errorf("Failed on TeardownPodNetwork for container ID %s: %v",
-			containerId, err)
-		return errors.Wrap(err, "del cmd: failed on tear down pod network")
+// tryDelWithPrevResult will try to process CNI delete request without IPAMD.
+// returns true if the del request is handled.
+func tryDelWithPrevResult(driverClient driver.NetworkAPIs, conf *NetConf, k8sArgs K8sArgs, contVethName string, log logger.Logger) (bool, error) {
+	// prevResult might not be available, if we are still using older cni spec < 0.4.0.
+	prevResult, ok := conf.PrevResult.(*current.Result)
+	if !ok {
+		return false, nil
 	}
-	return nil
+
+	dummyIfaceName := generateHostVethName(dummyVlanInterfacePrefix, string(k8sArgs.K8S_POD_NAMESPACE), string(k8sArgs.K8S_POD_NAME))
+	_, dummyIface, found := findCNIInterfaceByName(prevResult.Interfaces, dummyIfaceName)
```

Comment:
nit: I think this function can be named in generic fashion (i.e.,) `findInterfaceByName`

Reply:
i used `findCNIInterfaceByName` since `findInterfaceByName` is too generic as this function is specific to CNI.

maybe we can move it to some dedicated package such as cniUtils.go?

Reply:
Understand, we're looking for interfaces created by VPC CNI but the function actually can work for any interface name (in the result struct) (i.e.,) the function itself is implemented in generic fashion and hence the name suggestion. I do like the suggestion of making it an util function.

Reply:
Yes it would be good if we can move it to utils folder. I had created for this PR - https://github.com/aws/amazon-vpc-cni-k8s/pull/1726/files

Reply:
@achevuru @jayanthvn moved to a utils folder :D